### DISCUSSION TITLE
In-Built Content Protection

### DISCUSSION THREAD

**Discussion by dreaktr** [Tetrakern/fictioneer#55] [Ideas]
Is in-built content protection a feature you would consider?

For example, when people tried to copy the content they would be slammed with 'This content is protected' and if they enable Reader View Google extension, all they see is gibberish. I've seen sites hard-coded this into their site to protect their contents against scrappers, so I'm wondering if you are open to such suggestions.

**Comment by Tetrakern** [comment]
[No, because this doesn't work.](https://ux.stackexchange.com/questions/110855/how-big-a-deal-is-blocking-copy-pasting-on-a-website)

Let me explain. I have been annoyed by these "This content is protected" messages for quite a bit while trying to improve my theme, specifically looking for errors on sites using it. It takes me about 3 seconds to get around that, but still annoying. Here you can already see the issue: YOU CAN GET AROUND!

See, most of these "protections" rely on JavaScript to disable or intercept certain OS/Browser functions, such as text select or copy. That does a whole lotta crap if one just disables JavaScript, which is easily done. Suddenly, your protection is non-existent. Blocking me from open the console? Well, I can just open it elsewhere and navigate to the site. I can just copy the content straight out of the DOM and convert it back to plain text. And so on, and so forth.

What this especially does not stop is scrappers. Scrappers use bots to pull content from sites, which do not actually render the site or execute JavaScript in the first place. They just get the document and extract the content. Worst case, they can run an OCR to "read" the site. Provided you are worth the effort, of course. Not being valuable enough to bother is your best defense.

The only thing you achieve is annoying your readers. Same as with piracy protection in games, you only hit the honest customer and get laughed at by the rest. Any means that actually pose a challenge to content scrappers will severely impact your site's performance, usability, and reading experience. And they will still not stop them if they are determined.

Scripts can be blocked. Content can be decoded. Screens can be optically scanned.

You are not protected, you are inconvenient.

---

Now, I do have a few ideas to make the life of scrappers harder, such as hiding messages into the content "This content was stolen from" or outright SQL injections if they do not sanitize the content they steal. But even that is likely to fail and I'm not sure investing the time and effort is worth it.

**Reply by dreaktr** [reply]
Too bad this isn't something you would consider in doing. Nevertheless, thanks for answering!
